### PR TITLE
[CORL-2251] Fix sign in bug

### PR DESCRIPTION
**Pull request by marcushaddon**
## What does this PR do?
This PR addresses https://vmproduct.atlassian.net/browse/CORL-2251 by displaying validation errors only when a field has actually been "dirtied" or if the form as been unsuccessfully submitted (previously they would display as soon as a field lost focus)

## What changes to the GraphQL/Database Schema does this PR introduce?
None.

## How do I test this PR?
Navigate to the stream while not logged in. Click the "Sign in" button, and without typing anything into the email field, click the top half of any of Facebook/Google sign in buttons, and observe that the modal begins the OAuth flow, rather than sliding out from under your pointer.
 
## How do we deploy this PR?
No special considerations should have to be taken.

### CODE REVIEW

File: src/core/client/framework/lib/form/helpers.tsx [hunk 1]
```diff
@@ -97,11 +97,15 @@ export const formatStringList = (v: string[] | null) => {
...
 >;
 
-export const hasError = ({ touched, error, submitError }: FieldMeta) =>
-  touched && (error || submitError);
+export const hasError = ({
+  error,
+  submitError,
+  dirty,
+  submitFailed,
+}: FieldMeta) => (dirty || submitFailed) && (error || submitError);
```
Comment (alinhle) [inline src/core/client/framework/lib/form/helpers.tsx]: That's a nice improvement! Even better I think is leaving `dirty` out and only use `submitFailed && (error || submitError);`

**Review by alinhle — CHANGES_REQUESTED** [review]

File: src/core/client/framework/lib/form/helpers.tsx (hunk 1 above)
Comment (nick-funk) [reply src/core/client/framework/lib/form/helpers.tsx]: Yeah, this makes sense, no need to check for dirty. That seems like an overzealous "error" state haha.

Also, seconding that this is a nice fix @marcushaddon!

File: src/core/client/auth/views/SignIn/SignInWithEmail.tsx [hunk 2]
```diff
@@ -50,7 +50,7 @@ const SignInWithEmail: FunctionComponent<SignInWithEmailForm> = (props) => {
               />
             )}
             <div className={cn(CLASSES.login.field, styles.field)}>
-              <EmailField disabled={submitting} autofocus />
+              <EmailField disabled={submitting} />
```
Comment (nick-funk) [inline src/core/client/auth/views/SignIn/SignInWithEmail.tsx]: If you follow @cvle's suggestion and remove `dirty` from the `hasError` check, you can put this `autofocus` back in which improves keyboard usability by allowing folks to immediately start typing in the form when logging in with email.

**Review by nick-funk — DISMISSED** [review]

File: src/core/client/auth/views/SignIn/SignInWithEmail.tsx (hunk 2 above)
Comment (alinhle) [reply src/core/client/auth/views/SignIn/SignInWithEmail.tsx]: Thanks @nick-funk  I thought I put out a similar comment, but I guess I didn't click on submit :-P

Comment (marcushaddon) [reply src/core/client/auth/views/SignIn/SignInWithEmail.tsx]: ah i forgot to re-add that

File: src/core/client/framework/lib/form/helpers.tsx (hunk 1 above)
Comment (marcushaddon) [reply src/core/client/framework/lib/form/helpers.tsx]: this works perfecty!

**Review by alinhle — APPROVED** [review]

**Review by nick-funk — APPROVED** [review]

**Pull request merged**